src/Components/Artwork/GridItem.tsx

```diff
@@ -28,11 +29,55 @@ interface Props extends RelayProps, React.HTMLProps<ArtworkGridItemContainer> {
   }
 }
 
-class ArtworkGridItemContainer extends React.Component<Props, null> {
+interface State {
+  width: number
+  height: number
+}
+
+const IMAGE_QUALITY = 80
+
+class ArtworkGridItemContainer extends React.Component<Props, State> {
   static defaultProps = {
     useRelay: true,
   }
 
+  private image: HTMLImageElement = null
+
+  state = {
+    width: 0,
+    height: 0,
+  }
+
+  componentDidMount() {
+    const scale = window.devicePixelRatio
+    const width = this.image.width * scale
+    const height =
+      this.image.width * this.props.artwork.image.aspect_ratio * scale
```

Comment:
This is what's causing the 'fit' resizing to not work properly, (although 'width' resizing will work with these).

Since aspect ratio is generally defined as ratio of width to height (aka Width divided by Height), if you have the width and want to derive the height, you divide by the aspect ratio rather than multiply.

So, `height = width / aspect`. Then you can use 'fit' rather than 'width' in the resizing.